test/integration/inherited-entities/pm-variables.test.js

```diff
@@ -0,0 +1,307 @@
+                            script: {
+                                exec: 'console.log(pm.variables.toObject())'
+                            }
+                        },
+                        {
+                            listen: 'test',
+                            script: {
+                                exec: 'console.log(pm.variables.toObject())'
+                            }
+                        }
+                    ],
+                    request: {
+                        url: 'https://postman-echo.com/get?param={{key-1}}:{{key-2}}:{{key-3}}:{{key-4}}',
+                        auth: {
+                            type: 'bearer',
+                            bearer: {
+                                token: '{{key-1}}:{{key-2}}:{{key-3}}:{{key-4}}'
+                            }
+                        }
+                    }
+                }
+            }
+        };
+
+    describe('precedence', function () {
+        before(function (done) {
+            this.run(runOptions, function (err, results) {
+                testRun = results;
+
+                done(err);
+            });
+        });
+
+        it('must be honoured in scripts', function () {
+            var consoleLogs = testRun.console.getCall(0).args.slice(2)
+                .concat(testRun.console.getCall(1).args.slice(2))
+                .concat(testRun.console.getCall(2).args.slice(2))
+                .concat(testRun.console.getCall(3).args.slice(2));
+
+            consoleLogs.forEach(function (consoleLog) {
+                expect(consoleLog).to.eql({
+                    'key-1': 'global-value-1',
+                    'key-2': 'coll-value-2',
+                    'key-3': 'env-value-3',
+                    'key-4': 'data-value-4'
+                });
+            });
+        });
+
+        it('must be honoured in request URL', function () {
+            var url = testRun.request.getCall(0).args[3].url.toString(),
+                expectedParam = 'global-value-1:coll-value-2:env-value-3:data-value-4';
+
+            expect(url).to.be('https://postman-echo.com/get?param=' + expectedParam);
+        });
+
+        it('must be honoured in auth', function () {
+            var response = testRun.response.getCall(0).args[2],
+                expectedToken = 'global-value-1:coll-value-2:env-value-3:data-value-4';
+
+            expect(response.json().headers).to.have.property('authorization', 'Bearer ' + expectedToken);
+        });
+    });
+
+    describe('getters and setters', function () {
+        before(function (done) {
+            var clonedRunOptions = _.merge({}, runOptions, {
+                collection: {
+                    event: [
+                        {
+                            listen: 'prerequest',
+                            script: {
+                                exec: `
+                                // only run for 1st item
+                                if (pm.variables.get('key-1') !== 'modified-1') {
```

Comment:
@harryi3t can we try to use `unset` and avoid logic in the test fixtures to make it easy to comprehend?

Reply:
ok will try it out